### PR TITLE
feat: quick fix for disabling diagnostic code for line

### LINKED ISSUE #416: Quick-"fix" to all problems: ignore
Inspired by this, I suggest to add a quick-fix that can be applied to all problems found: Add an ignore.

image

image

image

For vscode-septic it would make sense to have two options available: `// noqa: xxx` and `{# noqa: xxx #}`.

### PR DESCRIPTION
**Pull request by vetlek**
Closes #416

### CODE REVIEW

File: server/src/language-service/codeActionProvider.ts [hunk 1]
```diff
@@ -3,57 +3,103 @@ import {
...
+                default:
+                    continue;
+            }
+        }
+        return codeActions;
     }
 }
 
-export interface CodeActionInsertEvr {
+enum CodaActionInsertType {
```
Comment (einarsi) [inline server/src/language-service/codeActionProvider.ts:83]: Typo? Cod/e/ActionInsertType

**Comment by einarsi** [comment]
Getting seemingly two warnings here:

image

Disabling W101 removes all warnings on the line. Disabling W501 leaves W101 as active warning.

**Review by einarsi — CHANGES_REQUESTED** [review]

**Comment by vetlek** [comment]
> Getting seemingly two warnings here:
> 
> image
> 
> Disabling W101 removes all warnings on the line. Disabling W501 leaves W101 as active warning.

W501 is still there, but the severity level for W501 in CalcPvr identifiers are set to Hint, which makes it small. It is still there when I test.

File: server/src/language-service/codeActionProvider.ts [hunk 2]
```diff
@@ -63,59 +102,216 @@ export function getCodeActionInsertEvr(
...
+}
+
+function createCodeActionInsertIgnoreComment(
+    code: string,
+    pos: Position,
+    diagnostics: Diagnostic[],
+    doc: ITextDocument
+): CodeAction[] {
+    let codeActionJinja = CodeAction.create(
+        `${code}: Disable for this line scg`
```
Comment (einarsi) [inline server/src/language-service/codeActionProvider.ts:236]:
```suggestion
        `${code}: Disable with {# noqa: ${code} #}`
```

File: server/src/language-service/codeActionProvider.ts [hunk 3]
```diff
@@ -63,59 +102,216 @@ export function getCodeActionInsertEvr(
...
+    );
+    codeActionJinja.diagnostics = diagnostics;
+    codeActionJinja.kind = CodeActionKind.QuickFix;
+    codeActionJinja.edit = createCodeActionInsertIgnoreCommentEdit(
+        `  {# noqa: ${code} #}`,
+        pos,
+        doc
+    );
+    let codeAction = CodeAction.create(`${code}: Disable for this line`);
```
Comment (einarsi) [inline server/src/language-service/codeActionProvider.ts:245]:
```suggestion
    let codeAction = CodeAction.create(`${code}: Disable with // noqa: ${code}`);
```

**Review by einarsi — APPROVED** [review]
Looks great! Made a suggestion to make the quick fix more clear that you can consider.

**Comment by einarsi** [comment]
> Disabling W101 removes all warnings on the line. Disabling W501 leaves W101 as active warning.

Indeed, I see it now.

**Pull request merged**